packages/headless/src/api/search/date/relative-date.ts

```diff
@@ -0,0 +1,167 @@
+import dayjs, {QUnitType} from 'dayjs';
+import {BooleanValue, NumberValue, Schema, StringValue} from '@coveo/bueno';
+import {formatDateForSearchApi} from './date-format';
+import utc from 'dayjs/plugin/utc';
+import quarterOfYear from 'dayjs/plugin/quarterOfYear';
+
+dayjs.extend(utc);
+dayjs.extend(quarterOfYear);
+
+/**
+ * The period to which to can set the date relative to.
+ */
+export type RelativeDatePeriod = 'past' | 'now' | 'future';
+const validRelativeDatePeriods: RelativeDatePeriod[] = [
+  'past',
+  'now',
+  'future',
+];
+
+/**
+ * The unit to which to can set the date relative to.
+ */
+export type RelativeDateUnit =
+  | 'minute'
+  | 'hour'
+  | 'day'
+  | 'week'
+  | 'month'
+  | 'quarter'
+  | 'year';
+const validRelativeDateUnits: RelativeDateUnit[] = [
+  'minute',
+  'hour',
+  'day',
+  'week',
+  'month',
+  'quarter',
+  'year',
+];
+
+/**
+ * Allows to define a date relative to the current moment.
+ */
+export interface RelativeDate {
+  /**
+   * The relative period of time.
+   */
+  period: RelativeDatePeriod;
+  /**
+   * The unit of time. When `period` is `now`, does not have to be defined.
+   */
+  unit?: RelativeDateUnit;
+  /**
+   * The amount of the `unit` of time. When `period` is `now`, does not have to be defined.
+   */
+  amount?: number;
+  /**
+   * If `true`, the date will be returned unshifted. If `false`, the date will be adjusted to UTC time.
+   *
+   * @defaultValue `false`
+   */
+  useLocalTime?: boolean;
+}
+
+const buildRelativeDateDefinition = (period: RelativeDatePeriod) => {
+  const isNow = period === 'now';
+  return {
+    amount: new NumberValue({required: !isNow, min: 1}),
+    unit: new StringValue({
+      required: !isNow,
+      constrainTo: validRelativeDateUnits,
+    }),
+    period: new StringValue({
+      required: true,
+      constrainTo: validRelativeDatePeriods,
+    }),
+    useLocalTime: new BooleanValue(),
+  };
+};
+
+function validateRelativeDate(relativeDate: RelativeDate) {
+  new Schema(buildRelativeDateDefinition(relativeDate.period)).validate(
+    relativeDate
+  );
+}
+
+/**
+ * Formats a `RelativeDate` object to a parsable string value.
+ * @param relativeDate The `RelativeDate` object.
+ */
+export function formatRelativeDate(relativeDate: RelativeDate) {
+  validateRelativeDate(relativeDate);
+  const {period, amount, unit, useLocalTime} = relativeDate;
+  const utc = useLocalTime ? '' : 'utc';
+
+  switch (period) {
+    case 'past':
+    case 'future':
+      return `${period}${amount}${unit}${utc}`;
+    case 'now':
+      return `${period}${utc}`;
```

Comment:
It would be good to add a default case. I would suggest making the `now` case the fallback/default.

Reply:
TS would warn if I didn't handle a string literal